scientific_library/tvb/simulator/models/reduced_wong_wang_exc_io_inh_i.py

```diff
@@ -0,0 +1,357 @@
+# When using The Virtual Brain for scientific publications, please cite it as follows:
+#
+#   Paula Sanz Leon, Stuart A. Knock, M. Marmaduke Woodman, Lia Domide,
+#   Jochen Mersmann, Anthony R. McIntosh, Viktor Jirsa (2013)
+#       The Virtual Brain: a simulator of primate brain network dynamics.
+#   Frontiers in Neuroinformatics (7:10. doi: 10.3389/fninf.2013.00010)
+
+"""
+Models based on Wong-Wang's work.
+
+"""
+
+from numba import guvectorize, float64
+from tvb.simulator.models.base import numpy, ModelNumbaDfun
+from tvb.basic.neotraits.api import NArray, Final, List, Range
+
+
+@guvectorize([(float64[:],)*19], '(n),(m)' + ',()'*16 + '->(n)', nopython=True)
+def _numba_update_non_state_variables(S, c, ae, be, de, wp, we, jn, re, ai, bi, di, wi, ji, ri, g, l, io, newS):
+    "Gufunc for reduced Wong-Wang model equations."
+
+    cc = g[0]*jn[0]*c[0]
+
+    jnSe = jn[0] * S[0]
+
+    if re[0] < 0.0:
+        x = wp[0]*jnSe - ji[0]*S[1] + we[0]*io[0] + cc
+        x = ae[0]*x - be[0]
+        h = x / (1 - numpy.exp(-de[0]*x))
+        S[2] = h
+
+    if ri[0] < 0.0:
+        x = jnSe - S[1] + wi[0]*io[0] + l[0]*cc
+        x = ai[0]*x - bi[0]
+        h = x / (1 - numpy.exp(-di[0]*x))
+        S[3] = h
+
+    newS[0] = S[0]
+    newS[1] = S[1]
+    newS[3] = S[2]
+    newS[4] = S[4]
+
+
+@guvectorize([(float64[:],)*6], '(n)' + ',()'*4 + '->(n)', nopython=True)
+def _numba_dfun(S, ge, te, gi, ti, dx):
+    "Gufunc for reduced Wong-Wang model equations."
+
+    dx[0] = - (S[0] / te[0]) + (1.0 - S[0]) * S[2] * ge[0]
+    dx[2] = 0.0
+
+    dx[1] = - (S[1] / ti[0]) + S[3] * gi[0]
+    dx[3] = 0.0
+
+
+class ReducedWongWangExcIOInhI(ModelNumbaDfun):
+    r"""
+    .. [WW_2006] Kong-Fatt Wong and Xiao-Jing Wang,  *A Recurrent Network
+                Mechanism of Time Integration in Perceptual Decisions*.
+                Journal of Neuroscience 26(4), 1314-1328, 2006.
+
+    .. [DPA_2014] Deco Gustavo, Ponce Alvarez Adrian, Patric Hagmann,
+                  Gian Luca Romani, Dante Mantini, and Maurizio Corbetta. *How Local
+                  Excitation–Inhibition Ratio Impacts the Whole Brain Dynamics*.
+                  The Journal of Neuroscience 34(23), 7886 –7898, 2014.
+
+
+
+    .. automethod:: ReducedWongWang.__init__
+
+    Equations taken from [DPA_2013]_ , page 11242
+
+    .. math::
+                 x_{ek}       &=   w_p\,J_N \, S_{ek} - J_iS_{ik} + W_eI_o + GJ_N \mathbf\Gamma(S_{ek}, S_{ej}, u_{kj}),\\
+                 H(x_{ek})    &=  \dfrac{a_ex_{ek}- b_e}{1 - \exp(-d_e(a_ex_{ek} -b_e))},\\
+                 \dot{S}_{ek} &= -\dfrac{S_{ek}}{\tau_e} + (1 - S_{ek}) \, \gammaH(x_{ek}) \,
+
+                 x_{ik}       &=   J_N \, S_{ek} - S_{ik} + W_iI_o + \lambdaGJ_N \mathbf\Gamma(S_{ik}, S_{ej}, u_{kj}),\\
+                 H(x_{ik})    &=  \dfrac{a_ix_{ik} - b_i}{1 - \exp(-d_i(a_ix_{ik} -b_i))},\\
+                 \dot{S}_{ik} &= -\dfrac{S_{ik}}{\tau_i} + \gamma_iH(x_{ik}) \,
+
+    """
+    _ui_name = "Reduced Wong-Wang"
```

Comment:
no ui_* attributes should exist in tvb-library 2.0

Reply:
Done. Removed.